cst438_gradebook/src/test/java/com/cst438/JunitTestGradeBookBackend.java

```diff
@@ -0,0 +1,302 @@
+		assertEquals(TEST_STUDENT_NAME, result.grades.get(0).name);
+		assertEquals("80", result.grades.get(0).grade);
+		
+	}
+	
+	@Test
+	public void deleteAssignment() throws Exception{
+		
+		MockHttpServletResponse response;
+		MockHttpServletResponse responseForSecondAssignment;
+		
+		//Creates new course
+		Course course = new Course();
+		course.setCourse_id(TEST_COURSE_ID);
+		course.setCourse_id(TEST_COURSE_ID);
+		course.setSemester(TEST_SEMESTER);
+		course.setYear(TEST_YEAR);
+		
+		//Creates an assignment with the variable NeedsGrading = 1
+		Assignment assignment = new Assignment();
+		assignment.setCourse(course);
+		assignment.setDueDate(new java.sql.Date(System.currentTimeMillis() - 7 * 24 * 60 * 60 * 1000));
+		assignment.setId(1);
+		assignment.setName("Assignment 1");
+		assignment.setNeedsGrading(1);
+		
+		//Returns data so that null values are not returned
+		given(assignmentRepository.findAssignmentById(1)).willReturn(assignment);
+		given(assignmentRepository.save(any())).willReturn(assignment);
+
+		//The delete REST API deletes the assignment with the Id of 1
+		response = mvc.perform(
+				MockMvcRequestBuilders
+			    .delete("/gradebook/1"))
+				.andReturn().getResponse();
+		
+		//Confirms that delete was called and it returned OK
+		assertEquals(200, response.getStatus());
+		verify(assignmentRepository).delete(any(Assignment.class));
+		
+		//Creates a new assignment with the variable NeedsGrading = 0
+		// Since it is 0 delete should not be performed. 
+		Assignment assignmentGraded = new Assignment();
+		assignmentGraded.setCourse(course);
+		assignmentGraded.setDueDate(new java.sql.Date(System.currentTimeMillis() - 7 * 24 * 60 * 60 * 1000));
+		assignmentGraded.setId(2);
+		assignmentGraded.setName("Assignment 1");
+		assignmentGraded.setNeedsGrading(0);
+		
+		//Returns data so that a null value is not returned
+		given(assignmentRepository.findAssignmentById(2)).willReturn(assignmentGraded);
+		given(assignmentRepository.save(any())).willReturn(assignmentGraded);
+		
+		//The delete REST API deletes the assignment with the Id of 2
+		responseForSecondAssignment = mvc.perform(
+				MockMvcRequestBuilders
+			    .delete("/gradebook/2"))
+				.andReturn().getResponse();
+		
+		//Since the assignment is already graded it cannot be deleted
+		//Therefore the response that should be returned is 400 
+		//Even though it returns an error, we must confirm that delete was called
+		assertEquals(400, responseForSecondAssignment.getStatus());
+		verify(assignmentRepository).delete(any(Assignment.class));
+	}
+		
+
+	
+	private static String asJsonString(final Object obj) {
+		try {
+
+			return new ObjectMapper().writeValueAsString(obj);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+	
+	private static <T> T fromJsonString(String str, Class<T> valueType) {
+		try {
+			return new ObjectMapper().readValue(str, valueType);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+}
```

Comment:
Overall Good work there are very few if any issues. The code is clean, the comments are well-written and concise. Leaves nothing to be desired.